112 provide descriptive database connection error messages

nilsoberg opened the pull request:
Add database connection message into code. Added database validation to scripts. Moved database connection to highest level (e.g. at script instead of buried inside modules).

nilsoberg committed:
Pass database handle to modules instead of database object

nilsoberg committed:
Update module API to support error messages and create POD

nilsoberg committed:
Remove unnecessary logic for database handle cache

nilsoberg committed:
Add database connection validation and error messages

nilsoberg committed:
Use updated database interface API for MySQL database connections

nilsoberg committed:
Add configuration file format documentation

nilsoberg committed:
Add missing file

rbdavid commented on lib/EFI/Database.pm:
```diff
@@ -83,89 +150,262 @@ sub tableExists {
+
+=head3 C<getError()>
+
+Returns an errors that were detected during connection.
```
"any errors"

rbdavid commented on lib/EFI/Database.pm:
```diff
@@ -83,89 +150,262 @@ sub tableExists {
+
+=head4 Returns
+
+An empty string of there were no errors, otherwise a non-empty string with
```
"... if there ..."

rbdavid commented on docs/source/lib/EFI/Database.pm.rst:
```diff
@@ -0,0 +1,220 @@
+
+::
+
+   # Perform $gnn computations and save data
+   my $gnn = new EFI::GNT::GNN(...);
```
Missing the EFI::Database specific synopsis code block.

rbdavid commented on docs/source/lib/EFI/Database.pm.rst:
```diff
@@ -0,0 +1,220 @@
+``getError()``
+~~~~~~~~~~~~~~
+
+Returns an errors that were detected during connection.
```
"any errors"

rbdavid commented on docs/source/lib/EFI/Database.pm.rst:
```diff
@@ -0,0 +1,220 @@
+Returns
+^^^^^^^
+
+An empty string of there were no errors, otherwise a non-empty string
```
"... if there..."

rbdavid commented on lib/EFI/Database.pm:
```diff
@@ -83,89 +150,262 @@ sub tableExists {
+
+=head2 SYNOPSIS
+
+    # Perform $gnn computations and save data
+    my $gnn = new EFI::GNT::GNN(...);
```
GNN stuff instead of EFI::Database code.

rbdavid reviewed (approved):
Looks good, just a few grammatical errors and a missing synopsis code block.

nilsoberg committed:
Clarify and organization documentation

- Fix typos
- Create reference section in docs
- Move efi.config file format documentation from EFI::Database Perl documentation to a separate file in docs/reference

nilsoberg committed:
Fix missing documentation update
